x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/evaluator/mapper/EvaluatorMapper.java

```diff
@@ -14,15 +14,17 @@
 import org.elasticsearch.xpack.esql.core.expression.Expression;
 import org.elasticsearch.xpack.esql.planner.Layout;
 
-import java.util.function.Function;
-
 import static org.elasticsearch.compute.data.BlockUtils.fromArrayRow;
 import static org.elasticsearch.compute.data.BlockUtils.toJavaObject;
 
 /**
  * Expressions that have a mapping to an {@link ExpressionEvaluator}.
  */
 public interface EvaluatorMapper {
+    interface ToEvaluator {
+        ExpressionEvaluator.Factory toEvaluator(Expression expression);
```

Comment:
bikeshed: I think the previous `apply` was easier on the eyes; `toEvaluator.toEvaluator(...)` looks a bit funny.

Reply:
Fair. I figured if we were adding another thing to the interface it'd be weird to still have the method called `apply`.

Reply:
I think `apply` looks nicer too. What will the other method do? If the `apply` is the primary thing to call, that still sounds right to me.

Reply:
:+1: